modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/subscription/AndesSubscriptionManager.java

```diff
@@ -274,6 +274,242 @@ private void removeLocalSubscriptionAndNotify(AndesSubscription subscription) th
+            ProtocolType protocolType, DestinationType destinationType,String subscriptionIdPattern,
+            boolean isExactMatchSubscriptionId){
+
+        Set<AndesSubscription> filteredSubscriptions = new HashSet<>();
+        String messageRouter = destinationType.getAndesMessageRouter();
+
+        if(isExactMatchSubscriptionId){
+            for (AndesSubscription inactiveSubscription : subscriptions) {
+                if(inactiveSubscription.getStorageQueue().getMessageRouter().getName().equals(messageRouter)
+                        && inactiveSubscription.getProtocolType().equals(protocolType)
+                        && inactiveSubscription.getSubscriptionId().equalsIgnoreCase(subscriptionIdPattern)){
+                    filteredSubscriptions.add(inactiveSubscription);
+                }
+            }
+        }else{
+            for (AndesSubscription inactiveSubscription : subscriptions) {
+                if(inactiveSubscription.getStorageQueue().getMessageRouter().getName().equals(messageRouter)
+                        && inactiveSubscription.getProtocolType().equals(protocolType)
+                        && inactiveSubscription.getSubscriptionId().toLowerCase()
+                        .contains(subscriptionIdPattern.toLowerCase())){
+                    filteredSubscriptions.add(inactiveSubscription);
+                }
+            }
+        }
+        return filteredSubscriptions;
+    }
+
+    /**
+     * filter active subscriptions by subscription id
+     *
+     * @param subscriptions subscription list for further filtering
+     * @param subscriptionIdPattern regex to match with ID of the subscriber
+     * @param isExactMatchSubscriptionId exact match of subscription id or not
+     * @return Set of subscriptions filtered according to search criteria
+     */
+    private Set<AndesSubscription> filterActiveSubscriptionsBySubscriptionId(Set<AndesSubscription> subscriptions,String
+            subscriptionIdPattern, boolean isExactMatchSubscriptionId){
+
+        Set<AndesSubscription> filteredSubscriptions = new HashSet<>();
+
+        if(isExactMatchSubscriptionId) {
+            for (AndesSubscription subscription : subscriptions) {
+                if (subscriptionIdPattern.equalsIgnoreCase(subscription.getSubscriptionId())) {
+                    filteredSubscriptions.add(subscription);
+                } else if (subscription.isDurable() && subscription.getStorageQueue().getMessageRouter().getName()
+                        .equals(AMQPUtils.TOPIC_EXCHANGE_NAME)) {
+                    if (subscriptionIdPattern.equalsIgnoreCase(((DurableTopicSubscriber) subscription).getClientID())) {
+                        filteredSubscriptions.add(subscription);
+                    }
+                }
+            }
+        } else{
+            for (AndesSubscription subscription : subscriptions) {
+                if (subscription.getSubscriptionId().toLowerCase().contains(subscriptionIdPattern.toLowerCase())) {
```

Comment:
This may not work in all the scenarios. The best practice is to use rg.apache.commons.lang3.StringUtils.containsIgnoreCase. 

ref: http://stackoverflow.com/questions/14018478/string-contains-ignore-case